internal/crossdock/server/oneway/echo.go

```diff
@@ -91,3 +86,25 @@ func (o *onewayHandler) callHome(ctx context.Context, callBackAddr string, body
 		panic(fmt.Sprintf("could not make call back to client: %s", err))
 	}
 }
+
+type thriftHandler struct {
```

Comment:
I had to add separate types because the thrift and protobuf function names conflict, it's basically the same for thrift as it is now.